test/java/seedu/addressbook/commands/DeleteCommandTest.java

```diff
@@ -0,0 +1,181 @@
+        assertInvalidIndex(-1, addressBook, listWithEveryone);
+        assertInvalidIndex(listWithEveryone.size() + 1, addressBook, listWithEveryone);
+    }
+
+    @Test
+    public void execute_validIndex_personIsDeleted() throws PersonNotFoundException {
+        assertDeleteSuccessful(1, addressBook, listWithSurnameDoe);
+        assertDeleteSuccessful(listWithSurnameDoe.size(), addressBook, listWithSurnameDoe);
+        
+        int middleIndex = (listWithSurnameDoe.size() / 2) + 1;
+        assertDeleteSuccessful(middleIndex, addressBook, listWithSurnameDoe);
+    }
+
+    /**
+     * Creates a new delete command.
+     * 
+     * @param targetVisibleIndex of the person that we want to delete
+     * @param addressBook to perform the delete operation on, must not be null
+     * @param displayList to get the selected person to delete, must not be null
+     */
+    public DeleteCommand createDeleteCommand(int targetVisibleIndex, AddressBook addressBook,
+            List<ReadOnlyPerson> displayList) {
+
+        assert addressBook != null;
+        assert displayList != null;
+
+        DeleteCommand command = new DeleteCommand(targetVisibleIndex);
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+
+    /**
+     * Executes the command, and checks that the execution was what we had expected.
+     */
+    public void assertCommandBehaviour(DeleteCommand deleteCommand, String expectedMessage,
+            AddressBook expectedAddressBook, AddressBook actualAddressBook) {
+
+        CommandResult result = deleteCommand.execute();
+
+        assertEquals(expectedMessage, result.feedbackToUser);
+        assertEquals(expectedAddressBook.getAllPersons(), actualAddressBook.getAllPersons());
+    }
+    
+    /**
+     * Given an invalid index, checks that the command reports the invalidity of such index.
+     * 
+     * @param addressBook to perform the delete operation on, must not be null
+     * @param displayList to get the selected person to delete, must not be null
+     */
+    public void assertInvalidIndex(int invalidVisibleIndex, AddressBook addressBook, 
+            List<ReadOnlyPerson> displayList) {
+
+        assert addressBook != null;
+        assert displayList != null;
+        
+        DeleteCommand command = createDeleteCommand(invalidVisibleIndex, addressBook, displayList);
+
+        assertCommandBehaviour(command, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX, addressBook,
+                addressBook);
+    }
+    
+    /**
+     * Selects a person in the display list using the index, that does not exist in the address book, and 
+     * checks that the command reports that the person is not in the address book.
+     */
+    public void assertNoSuchPerson(int visibleIndex, AddressBook addressBook, 
+            List<ReadOnlyPerson> displayList) {
+
+        assert addressBook != null;
+        assert displayList != null;
```

Comment:
Perhaps no need for asserts as these are test methods. The test itself will fail if null is passed as parameters, right?